app/src/main/java/com/example/mac/chartr/activities/RegisterActivity.java

```diff
@@ -208,23 +291,41 @@ private void initSignup() {
     private void initPhone() {
         phone = (EditText) findViewById(R.id.editTextRegPhone);
         phone.addTextChangedListener(new TextWatcher() {
+            private MaskWatcher maskWatcher = new MaskWatcher("+##########");
+
             @Override
             public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                maskWatcher.beforeTextChanged(s, start, count, after);
+
                 if (s.length() == 0) {
                     TextView label = (TextView) findViewById(R.id.textViewRegPhoneLabel);
-                    label.setText(phone.getHint() + " with country code and no seperators");
+                    label.setText("Phone number with country code\n(example: +18883334444)");
                     phone.setBackground(getDrawable(R.drawable.text_border_selector));
                 }
             }
 
             @Override
             public void onTextChanged(CharSequence s, int start, int before, int count) {
-                TextView label = (TextView) findViewById(R.id.textViewRegPhoneMessage);
-                label.setText("");
+                maskWatcher.onTextChanged(s, start, before, count);
+
+                TextView message = (TextView) findViewById(R.id.textViewRegPhoneMessage);
+
+                Pattern pattern =
+                        Pattern.compile("^\\+1([1-9])(\\d{9})");
+                Matcher matcher = pattern.matcher(phone.getText().toString());
+
+                if (matcher.find() || (phone.getText().toString().length() > 2
+                        && phone.getText().toString().charAt(1) != '1')) {
+                    message.setText("");
+                } else {
+                    message.setText("Invalid phone number");
+                }
             }
```

Comment:
I really don't like how the phone input looks, is it ok if I change the phone related input so that it shows up as either:
+1-222-333-4444 or
+1 (222) 333-4444
in the input?  If so, which do you prefer?

Reply:
I agree that the current input does look rather bad. I would say +1 (222) 333-4444 is the ideal format. However, we'd just have to modify the regex for that. But honestly, that's not a complex pattern to recognize. Also look into the Mask Watcher which can do some of the auto formatting.

On thing though: the whole 
"+1 (222) 333-4444" is an American format, so the one thing is that it wouldn't really support the international formats. Currently, I'm doing a regex and checking that will allow different international format (only enforcing a specific US format if it starts with a 1). Now, we don't necessarily need to support all country phone numbers, but it's something to consider.

Reply:
I can change it so that it enforces the phone format the same way as it does the date.  It doesn't look too complicated to do if we stick with one format.

Reply:
Since we're assuming the length is 11 characters anyway, I'll just use the +1 (222) 333-4444 format for now.